buildspec/release/70checkmarketplace.yml

```diff
@@ -22,27 +22,33 @@ phases:
     build:
         commands:
             - VERSION=$(node -e "console.log(require('./packages/${TARGET_EXTENSION}/package.json').version);")
-            # get extension name
+            # get extension name, if in beta, use some hard-coded recent version
             - |
                 if [ "${TARGET_EXTENSION}" = "amazonq" ]; then
                   extension_name="amazonwebservices.amazon-q-vscode"
+                  [ "$STAGE" != "prod" ] && VERSION="1.43.0" || true
                 elif [ "${TARGET_EXTENSION}" = "toolkit" ]; then
                   extension_name="amazonwebservices.aws-toolkit-vscode"
+                  [ "$STAGE" != "prod" ] && VERSION="3.42.0" || true
                 else
                   echo checkmarketplace: "Unknown TARGET_EXTENSION: ${TARGET_EXTENSION}"
                   exit 1
                 fi
-            # keep reinstalling the extension until the desired version is updated. Otherwise fail on codebuild timeout (1 hour).
+            - |
+                if [ "$STAGE" != "prod" ]; then
+                  echo "checkmarketplace: Non-production stage detected. Installing hardcoded version '${VERSION}'."
+                fi
+            # keep installing the desired extension version until successful. Otherwise fail on codebuild timeout (1 hour).
             - |
                 while true; do
                   code --uninstall-extension "${extension_name}" --no-sandbox --user-data-dir /tmp/vscode
-                  code --install-extension ${extension_name} --no-sandbox --user-data-dir /tmp/vscode
+                  code --install-extension "${extension_name}@${VERSION}" --no-sandbox --user-data-dir /tmp/vscode
```

Comment:
`code` doesn't fail if the newly-published version isn't found yet? If it fails, we need `|| true` here. Otherwise the `while` loop will exit.

Reply:
```suggestion
                  code --install-extension "${extension_name}@${VERSION}" --no-sandbox --user-data-dir /tmp/vscode || true
```